File: 2015/Day2/day2.cpp
#include <iostream>
#include <fstream>
#include <algorithm>

using namespace std;

int main()
{
    ifstream inData;
    char ch;
    int l = 0, w = 0, h = 0;
    long tarea = 0, tribbon = 0;

    inData.open("input.txt");

    while (inData)
    {
        inData >> l;
        inData >> ch;
        inData >> w;
        inData >> ch;
        inData >> h;

        tarea += 2 * l * w + 2 * w * h + 2 * l * h;

        int sides[] = {l , w, h};
        sort(sides, sides + 3);

        tarea += sides[0] * sides[1];

        l = 0;
        w = 0;
        h = 0;

        tribbon += 2 * (sides[0] + sides[1]) + (sides[0] * sides[1] * sides[2]);
    }

    cout << tarea << " " << tribbon;
}

File: 2015/Day2/day2.py
from re import L


def main():
    with open("/home/ahmed/Documents/Advent-Of-Code/2015/Day2/input.txt") as input:
        lines = input.readlines()

        t_area, t_ribbon = 0, 0
        for line in lines:
            l, w, h = map(int, line[:-1].split("x"))

            t_area += 2 * l * w + 2 * w * h + 2 * l * h
            
            sides = [l, w ,h]
            sides.sort()

            t_area += sides[0] * sides[1]

            t_ribbon += 2 * (sides[0] + sides[1]) + (sides[0] * sides[1] * sides[2])

    print(t_area, "  ", t_ribbon)


main()